e2e/LongHaul/service/HubEvents.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+
+namespace Microsoft.Azure.Devices.LongHaul.Service
+{
+    internal class HubEvents
+    {
+        private readonly IotHub _iotHub;
+
+        internal HubEvents(IotHub serviceClient)
+        {
+            _iotHub = serviceClient;
+        }
+
+        internal async Task RunAsync(CancellationToken ct)
+        {
+            var readEventOptions = new ReadEventOptions { MaximumWaitTime = TimeSpan.FromSeconds(3) };
+            string eventHubConnectionString = await _iotHub.GetEventHubCompatibleConnectionStringAsync(ct);
+            await using var eventHubClient = new EventHubConsumerClient(
+                "$Default",
+                eventHubConnectionString);
+
+            while (!ct.IsCancellationRequested)
+            {
+                await foreach (PartitionEvent partitionEvent in eventHubClient.ReadEventsAsync(readEventOptions, ct))
+                {
+                    if (partitionEvent.Data == null)
+                    {
+                        await Task.Delay(5000, ct).ConfigureAwait(false);
+                        break;
+                    }
+
+                    try
+                    {
+                        DeviceEventSystemProperties eventMetadata = JsonSerializer.Deserialize<DeviceEventSystemProperties>(
+                            JsonSerializer.Serialize(partitionEvent.Data.SystemProperties));
+
+                        switch (eventMetadata.MessageSource)
+                        {
+                            case DeviceEventMessageSource.DeviceConnectionStateEvents:
+                                DeviceEventProperties deviceEvent = JsonSerializer.Deserialize<DeviceEventProperties>(
+                                    JsonSerializer.Serialize(partitionEvent.Data.Properties));
+                                TimeSpan timeSince = DateTimeOffset.UtcNow - deviceEvent.OperationOnUtc;
+                                if (timeSince.TotalSeconds < 30)
+                                {
+                                    Console.WriteLine($"{deviceEvent.HubName}/{deviceEvent.DeviceId} has event {deviceEvent.OperationType} at {deviceEvent.OperationOnUtc.LocalDateTime}, {timeSince} ago.");
```

Comment:
How about logging this with `logger.Trace` instead of console output? Same for below.